pkg/metrics/metrics.go

```diff
@@ -232,6 +241,29 @@ func (mc *MetricsConfig) Validate(ctx context.Context) error {
 	return nil
 }
 
+// removeStaleResources deletes any stale resources that could be left behind during the
+// upgrade process due to the renaming of resources between current and previous operator version
+func (mc *MetricsConfig) removeStaleResources(ctx context.Context) error {
+	if _, err := mc.CoreV1().Services(mc.namespace).Get(ctx, staleResourceName, metav1.GetOptions{}); err == nil {
+		if err = mc.CoreV1().Services(mc.namespace).Delete(ctx, staleResourceName, metav1.DeleteOptions{}); err != nil {
+			return errors.Wrap(err, "could not delete stale service object")
```

Comment:
Please make a best effort to clean up all stale resources rather than return on the first error encountered. I also think this function should not return an error bur rather log and send events.